compiler-cli/src/fs.rs

```diff
@@ -79,6 +79,20 @@ impl gleam_core::io::FileSystemReader for ProjectIO {
                 .collect()
         })
     }
+
+    fn erlang_files(&self, dir: &Path) -> Box<dyn Iterator<Item = PathBuf>> {
```

Comment:
This will not catch any files that are not `.erl` files. For example `.hrl`, `.mjs`, metadata and fingerprint files. Rather than try and find them all after the fact I think we should store on disc a record of all the files written and diff that against the files written in the latest compilation.

Reply:
Do you mean record of all the files written under function perform_codegen and encode_and_write_metadata right?

<img width="512" alt="Screen Shot 2565-04-05 at 22 09 28" src="https://user-images.githubusercontent.com/11692854/161761056-3ed027ad-17ca-4ce5-b793-790120aff83a.png">